cpmpy/solvers/pysat.py

```diff
@@ -243,40 +253,41 @@ def _post_constraint(self, cpm_expr):
                 raise NotImplementedError(
                     f"Automatic conversion of Operator {cpm_expr} to CNF not yet supported, please report on github.")
         elif isinstance(cpm_expr, Comparison):
+            left = cpm_expr.args[0] # left-hand side, sum/wsum/mul
+            bound = cpm_expr.args[1] # right-hand side, constant
+
             # only handle cardinality encodings (for now)
-            if isinstance(cpm_expr.args[0], Operator) and cpm_expr.args[0].name == "sum" and all(
-                    isinstance(v, _BoolVarImpl) for v in cpm_expr.args[0].args):
-                lits = self.solver_vars(cpm_expr.args[0].args)
-                bound = cpm_expr.args[1]
+            if isinstance(left, Operator) and left.name == "sum" and all(isinstance(v, _BoolVarImpl) for v in left.args):
+                lits = self.solver_vars(left.args)
                 if not is_int(bound):
                     raise NotImplementedError(f"PySAT sum: rhs `{bound}` type {type(bound)} not supported")
```

Comment:
Ideally, this would be something like `NotSupportedError`, as we cannot implement it :)

Reply:
actually OK to throw NotImplementedError...

e.g. if rhs is a sum, then the entirety is a linear constraint and if some transformation (see below, eg does linearize do this?) write all linear equalities in some normal form then it would be supported...